Microsoft.Toolkit.Uwp.UI/Extensions/VisualExtensions.cs

```diff
@@ -13,602 +10,239 @@
-        public static void SetSize(DependencyObject obj, string value)
+        /// <param name="element">The input <see cref="UIElement"/> instance.</param>
+        /// <param name="value">The <see cref="string"/> representation of the <see cref="Vector2"/> value to be set.</param>
+        public static void SetSize(UIElement element, string value)
         {
-            if (!DesignTimeHelpers.IsRunningInLegacyDesignerMode && obj is UIElement element)
-            {
-                SetSizeForElement(value, element);
-            }
-
-            obj.SetValue(SizeProperty, value);
+            ElementCompositionPreview.GetElementVisual(element).Size = value.ToVector2();
         }
 
         /// <summary>
-        /// Gets the <see cref="Visual.CenterPoint"/> of the <see cref="UIElement"/> normalized between 0.0 and 1.0
-        /// is centered even when the visual is resized
+        /// Identifies the "NormalizedCenterPoint" attached property.
         /// </summary>
-        /// <param name="obj">The <see cref="UIElement"/></param>
-        /// <returns>a string representing Vector2 as the normalized <see cref="Visual.CenterPoint"/></returns>
-        public static string GetNormalizedCenterPoint(DependencyObject obj)
-        {
-            return (string)obj.GetValue(NormalizedCenterPointProperty);
-        }
+        public static readonly DependencyProperty NormalizedCenterPointProperty = DependencyProperty.RegisterAttached(
+            "NormalizedCenterPoint",
+            typeof(string),
+            typeof(VisualExtensions),
+            new PropertyMetadata(false, OnNormalizedCenterPointChanged));
 
         /// <summary>
-        /// Sets the normalized <see cref="Visual.CenterPoint"/> of the <see cref="UIElement"/>
-        /// is centered even when the visual is resized
+        /// Gets the <see cref="Visual.CenterPoint"/> of a <see cref="UIElement"/> instance, normalized between 0.0 and 1.0.
         /// </summary>
-        /// <param name="obj">The <see cref="UIElement"/></param>
-        /// <param name="value">A string representing a Vector2 normalized between 0.0 and 1.0</param>
-        public static void SetNormalizedCenterPoint(DependencyObject obj, string value)
+        /// <param name="element">The input <see cref="UIElement"/> instance.</param>
+        /// <returns>a <see cref="Vector2"/> representation of the normalized <see cref="Visual.CenterPoint"/> property for <paramref name="element"/>.</returns>
+        public static string GetNormalizedCenterPoint(DependencyObject element)
         {
-            obj.SetValue(NormalizedCenterPointProperty, value);
+            return (string)element.GetValue(NormalizedCenterPointProperty);
         }
 
         /// <summary>
-        /// Identifies the AnchorPointProperty attached property.
-        /// </summary>
-        public static readonly DependencyProperty AnchorPointProperty =
```

Comment:
I was just thinking, now that we don't have the explicit registrations, does that mean that bindings won't work with these anymore to update the value???

Looking at it now, I think that's my only concern.

Reply:
Yup, just checked as well, this change makes the properties only work with either `{x:Bind}` or a static assignment from XAML, but both old bindings and eg. setters in a visual state don't work (I guess the same applies to the title bar extensions too then). Do you want me to reintroduce the backing properties to keep the support in these scenarios? 🤔

Reply:
Yeah, I'm not sure if anyone would have been updating these (at least maybe with a VSM) to do some basic options on visuals. Though I know our new animations package provides the ability to make more enriched scenarios, these are in more of a base package helper. So I think supporting more rich expressiveness with them would be important.

Can we kind of mash both worlds together or do we have to keep a backing field effectively to support that?

Reply:
The main issue is that all dynamic XAML features such as classic bindings, visual state setters, etc. all just try to match for a `DependencyProperty` with the right name for whatever attached property is being used, despite the various getter/setter methods being available. So not really seeing a way to make those work without reintroducing those backing properties unfortunately. We can maybe reduce the overhead just a bit by tweaking the code, removing those unnecessary designer checks etc., but as things stand I don't think we can avoid reintroducing the properties themselves if we want to support these other scenarios 😕